src/main/asciidoc/repositories.adoc

```diff
@@ -649,6 +649,39 @@ If you use namespace configuration, the repository infrastructure tries to autod
 
 The first configuration example will try to look up a class `com.acme.repository.UserRepositoryImpl` to act as custom repository implementation, whereas the second example will try to lookup `com.acme.repository.UserRepositoryFooBar`.
 
+===== Resolution of ambiguity
+
+If multiple implementations with matching class names get found in different packages Spring Data uses the bean names to identify the correct on to use.
+
+Given the following two custom implementations for the `UserRepository` introduced above the first implementation will get picked. It's bean name is `userRepositoryImpl` matches that of the repository interface (`userRepository`) plus the postfix `Impl`.
+
+.Resolution of amibiguous implementations
+====
+[source, java]
+----
+package com.acme.impl.one;
+
+class UserRepositoryImpl implements UserRepositoryCustom {
```

Comment:
I think we can even leave away the method declarations here. Should we describe how one would get the second implementation wired?